examples/cndpfwd/main.c

```diff
@@ -404,11 +411,38 @@ _thread_cleanup(jcfg_info_t *j __cne_unused, void *obj, void *arg, int idx __cne
         }
         if (ret < 0)
             CNE_ERR("port_close() returned error\n");
+    }
+    return 0;
+}
 
-        if (lport->umem) {
-            int i;
+static int
+_check_thread_quit(jcfg_info_t *j __cne_unused, void *obj, void *arg __cne_unused, int idx)
+{
+    jcfg_thd_t *thd = obj;
+
+    /* Make sure worker threads are done. Ignore main thread (idx=0) */
+    while (thd->quit != THD_DONE && idx > 0)
```

Comment:
I was looking at this forever loop and I was thinking we could add a usleep(10000); or some value to reduce the lcore utilization when it is stuck on this while loop. Plus we could do a timeout here too;

```
#define TIMEOUT_VALUE 1000 /* Number of times to wait for each usleep() time */
uint32_t timo = TIMEOUT_VALUE;

while(timo-- && thd->quit != THD_DONE && idx > 0)
    usleep(10000); /* 10ms */

if (timo < 0)
    return -1;

```
Just a thought as hanging is not good I would prefer it timeout then hang for some reason.

Need to update each place we can hang. As the code it common between examples it may exist in other examples too.

Reply:
> I was looking at this forever loop and I was thinking we could add a usleep(10000); or some value to reduce the lcore utilization when it is stuck on this while loop. Plus we could do a timeout here too;
> 
> ```
> #define TIMEOUT_VALUE 1000 /* Number of times to wait for each usleep() time */
> uint32_t timo = TIMEOUT_VALUE;
> 
> while(timo-- && thd->quit != THD_DONE && idx > 0)
>     usleep(10000); /* 10ms */
> 
> if (timo < 0)
>>>>>>>>>Isn't this always false since uint32_t?
>     return -1;
 >>>>>>>>>is this return value being used?
> ```
> 
> Just a thought as hanging is not good I would prefer it timeout then hang for some reason.
> 
> Need to update each place we can hang. As the code it common between examples it may exist in other examples too.

Reply:
Yes,  you are correct it should 'int timo = TIMEOUT_VALUE;' or we could move the -- to --timo and then test for zero below.

Reply:
Done